src/AzureAppConfigurationImpl.ts

```diff
@@ -74,6 +117,68 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
         }
     }
 
+    /**
+     * Refresh the configuration store.
+     */
+    public async refresh(): Promise<void> {
+        if (!this.refreshEnabled) {
+            return Promise.resolve();
+        }
+
+        // if still within refresh interval/backoff, return
+        if (this.refreshTimer.isDuringBackoff()) {

```

Comment:
How about naming the method something like `canRefresh()`? That covers both backoff and normal refresh interval wait period. Then we can check `if (!this.refreshTimer.canRefresh())`

Reply:
`canRefresh` looks good. 

BTW I used to refer to `needs_refresh` from python, which is a little bit unclear to me, because it doesn't really "need" a refresh, it stands for whether a real refresh is allowed.

Reply:
updated in 5b3ab06

Reply:
In Python, the variable `need_refresh` is used to track if any sentinel keys have changed on the server. You have a similar variable which makes sense because we are tracking if configuration needs to be refreshed after a server call has already been made.

Reply:
I meant [RefreshTimer.needs_refresh](https://github.com/Azure/azure-sdk-for-python/blob/98274a8e4684d868471654cfe92469d452316428/sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py#L408-L410).